pinot-core/src/test/java/org/apache/pinot/core/util/SegmentProcessorAvroUtilsTest.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import org.apache.avro.generic.GenericData;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.data.readers.GenericRow;
+import org.testng.SkipException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+public class SegmentProcessorAvroUtilsTest {
+
+
+  @DataProvider
+  public static Object[][] getTypeConversion() {
+    return new Object[][] {
+        new Object[] {FieldSpec.DataType.INT, org.apache.avro.Schema.Type.INT},
+        new Object[] {FieldSpec.DataType.LONG, org.apache.avro.Schema.Type.LONG},
+        new Object[] {FieldSpec.DataType.FLOAT, org.apache.avro.Schema.Type.FLOAT},
+        new Object[] {FieldSpec.DataType.DOUBLE, org.apache.avro.Schema.Type.DOUBLE},
+        new Object[] {FieldSpec.DataType.STRING, org.apache.avro.Schema.Type.STRING},
+        new Object[] {FieldSpec.DataType.BIG_DECIMAL, org.apache.avro.Schema.Type.STRING},
+        new Object[] {FieldSpec.DataType.BYTES, org.apache.avro.Schema.Type.BYTES},
+        new Object[] {FieldSpec.DataType.BOOLEAN, org.apache.avro.Schema.Type.INT},
+        new Object[] {FieldSpec.DataType.JSON, org.apache.avro.Schema.Type.STRING},
+        new Object[] {FieldSpec.DataType.TIMESTAMP, org.apache.avro.Schema.Type.LONG}
+    };
+  }
+
```

Comment:
Would it also make sense to test conversion of null (of nullable-column, of course) ?